File: src/app/Class/user/UserSessionStorageData.ts
export class UserSessionStorageData {

    private userName: string;
    private email: string;
    private roles: string[];

    constructor(obj?: any) {
        this.userName = (obj && obj.userName) || '';
        this.email = (obj && obj.email) || '';
        this.roles = (obj && obj.roles) || null;
    }

    /** Getters */

    getUserName(): string {
        return this.userName;
    }
    getEmail(): string {
        return this.email;
    }

    getRoles(): string[] {
        return this.roles;
    }

    /** Setters */

    setUserName(value: string) {
        this.userName = value;
    }

    setEmail(value: string) {
        this.email = value;
    }

    setRoles(value: string[]) {
        this.roles = value;
    }

}

File: src/main/java/MusicCube/user/userSessionStorageData.java
package MusicCube.user;

import MusicCube.entities.Role;
import MusicCube.entities.User;

import java.util.HashSet;
import java.util.Set;

public class userSessionStorageData {

    private String userName;
    private String email;
    private Set<Role> roles = new HashSet<>();

    // Constructors

    public userSessionStorageData() {
    }

    public userSessionStorageData(User user) {
        this.userName = user.getUserName();
        this.email = user.getEmail();
        this.roles = user.getRoles();
    }

    public userSessionStorageData(String userName, String email, Set<Role> roles) {
        this.userName = userName;
        this.email = email;
        this.roles = roles;
    }

    // Getters and Setters

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public Set<Role> getRoles() {
        return roles;
    }

    public void setRoles(Set<Role> roles) {
        this.roles = roles;
    }

}
